File: java/assignment/src/main/java/com/exception/SystemException.java
package com.exception;

public class SystemException extends RuntimeException {
    private static final long serialVersionUID = 1L;
    private int code;
    private String message;

    public SystemException(int code, String msg) {
        super(msg);
        this.code = code;
        this.message = msg;
    }

    public int getCode() {
        return this.code;
    }

    public String getMessage() {
        return this.message;
    }
}

File: fullstack/common/exception/system-exception.ts
import { getErrorMessage } from "./exception-const";

export class SystemException extends Error {
  code: number;
  msg: string;
  err: Error | undefined;
  constructor(errCode: number, errMsg?: string[], err?: Error) {
    let msg: string = getErrorMessage(errCode, errMsg);
    super(msg);
    this.code = errCode;
    this.msg = msg;
    if (err) {
      this.err = err;
    }
  }
}
